lib/components/ScheduleSelector/components/HourPicker/HourPicker.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useMemo, useCallback } from 'react'
+import { ToggleButton } from '../../../index'
+
+import './hourPicker.scss'
+
+interface TimeOption {
+  value: string
+  label: string
+}
+
+interface HourPickerProps {
+  hours: string
+  minuteOffset: number
+  selectedHours: string[]
+  onChange: (hours: string[]) => void
+  isDisabled?: boolean
+}
+
+const HourPicker: React.FC<HourPickerProps> = ({
+  hours,
+  minuteOffset,
+  selectedHours,
+  onChange,
+  isDisabled
+}) => {
+  const parseHours = useMemo(() => {
+    return (hours: string): string[] =>
+      hours?.split(',').map((hour) => hour.trim()) || []
+  }, [])
+
+  const parsedHours = useMemo(() => parseHours(hours), [hours, parseHours])
+
+  const generateHourlyOptions = useCallback(
+    (startHour: number, offset: number): TimeOption[] => {
+      const options: TimeOption[] = []
+      for (let i = 0; i < 24; i++) {
+        const hour = (startHour + i) % 24
+        const hourString = hour.toString().padStart(2, '0')
+        const minuteString = offset.toString().padStart(2, '0')
```

Comment:
suggesting to add '0' to a const as well since it is repeated several times

Reply:
We have it in other locations on the code base, for example `IpSubnetTextBox`, and other places, in my opinion, it is redundant to create constant for "ZERO", as it will never be something else, or not enough readable. but if we want to change it , we need to change it everywhere

Reply:
`EMPTY_STRING` also can never be something else, but I think it's a good practice to have strings that are repeated in a const. But agree about everywhere - that's for sure.

Reply:
But can definitely be done later as a refactoring thing